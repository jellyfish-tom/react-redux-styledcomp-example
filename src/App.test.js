import React from 'react';
import Calculations from './components/calculations';
import LoanCalculator from './components/loan-calculator';
import LogoContainer from './components/logo-container';
import renderer from 'react-test-renderer';

const loan = {
  loanDurationInMonths: 6,
  loanGBPAmount: 1000,
  totalInterest: { currency: 'GBP', amount: 0 },
  monthlyPayment: { currency: 'GBP', amount: 0 },
  actionPending: false,
  updated: true,
  error: null,
};

test('Calculations component matches snapshot', () => {
  const component = renderer.create(
    <Calculations
      onLoanAmountChange={data => {}}
      onLoanDurationChange={data => {}}
      calculateLoan={() => {}}
      loan={loan}
    />,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('LoanCalculator component matches snapshot', () => {
  const component = renderer.create(<LoanCalculator loan={loan} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('LogoContainer component matches snapshot', () => {
  const component = renderer.create(<LogoContainer />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
