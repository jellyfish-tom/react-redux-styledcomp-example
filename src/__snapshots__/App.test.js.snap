// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculations component matches snapshot 1`] = `
<div
  className="sc-AxjAm gUIIxw"
>
  <div>
    <span>
      Total interest:
    </span>
    <span>
      0 GBP
    </span>
  </div>
  <div>
    <span>
      Your montly cost:
    </span>
    <span>
      0 GBP 
    </span>
  </div>
</div>
`;

exports[`LoanCalculator component matches snapshot 1`] = `
<div
  className="sc-AxhCb jQUMEI"
>
  <div
    className="sc-AxirZ fMSpXb"
  >
    <h3>
      Calculated for you with love by
    </h3>
    <img
      alt="logo"
      src=""
    />
    <h4>
      <span>
        Loan
      </span>
      <span>
        Calculator
      </span>
    </h4>
  </div>
  <div
    className="sc-AxhUy elhXSh"
  >
    <span
      className="sc-AxgMl czpDHs"
    >
      <span>
        How much would you like to borrow:
      </span>
      <span>
        1000
         GBP
      </span>
    </span>
    <div
      className="sc-AxheI kfpleL"
    >
      <span>
        1000 GBP
      </span>
      <div
        className="rc-slider sc-AxiKw bOTXee"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#ffcbc9",
              "height": 8,
            }
          }
        />
        <div
          className="rc-slider-track"
          style={
            Object {
              "backgroundColor": "#6559fb",
              "height": 8,
              "left": "0%",
              "right": "auto",
              "width": "0%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={5000}
          aria-valuemin={1000}
          aria-valuenow={1000}
          className="rc-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="slider"
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#6559fb",
              "height": 20,
              "left": "0%",
              "marginLeft": 0,
              "marginTop": -6,
              "right": "auto",
              "transform": "translateX(-50%)",
              "width": 20,
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span>
        5000 GBP
      </span>
    </div>
  </div>
  <div
    className="sc-AxhUy elhXSh"
  >
    <span
      className="sc-AxgMl czpDHs"
    >
      <span>
        For how long:
      </span>
      <span>
        6
         months
      </span>
    </span>
    <div
      className="sc-AxheI kfpleL"
    >
      <span>
        6 months
      </span>
      <div
        className="rc-slider sc-AxiKw bOTXee"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#ffcbc9",
              "height": 8,
            }
          }
        />
        <div
          className="rc-slider-track"
          style={
            Object {
              "backgroundColor": "#6559fb",
              "height": 8,
              "left": "0%",
              "right": "auto",
              "width": "0%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={36}
          aria-valuemin={6}
          aria-valuenow={6}
          className="rc-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="slider"
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#6559fb",
              "height": 20,
              "left": "0%",
              "marginLeft": 0,
              "marginTop": -6,
              "right": "auto",
              "transform": "translateX(-50%)",
              "width": 20,
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span>
        36 months
      </span>
    </div>
  </div>
  <div
    className="sc-AxjAm gUIIxw"
  >
    <div>
      <span>
        Total interest:
      </span>
      <span>
        0 GBP
      </span>
    </div>
    <div>
      <span>
        Your montly cost:
      </span>
      <span>
        0 GBP 
      </span>
    </div>
  </div>
</div>
`;

exports[`LogoContainer component matches snapshot 1`] = `
<div
  className="sc-AxirZ fMSpXb"
>
  <h3>
    Calculated for you with love by
  </h3>
  <img
    alt="logo"
    src=""
  />
  <h4>
    <span>
      Loan
    </span>
    <span>
      Calculator
    </span>
  </h4>
</div>
`;
